keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1279,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
-- Empty `ResourceClassName` and nil `ParametersRef`: this is a claim which requests
-  no resources. Such a claim can always be allocated with an empty result. Allowing
-  this simplifies code generators which dynamically fill in the resource requests
-  because they are allowed to generate an empty claim.
-- Non-empty `ResourceClassName`, nil `ParametersRef`, nil
-  `ResourceClass.DefaultClaimParametersRef`: this is handled the same way, the
-  only difference is that the cluster admin has decided that such claims need
-  no resources by not providing default parameters.
-
-There is no default ResourceClass. If that is desirable, then it can be
-implemented with a mutating and/or admission webhook.
+    // Future extension, ignored by older schedulers. This is fine because scoring
+    // allows users to define a preference, without making it a hard requirement.
+    //
+    //
+    // Score *SomeScoringStruct
 
-```
-// ResourceClaimStatus tracks whether the resource has been allocated and what
-// the resulting attributes are.
-type ResourceClaimStatus struct {
-    // Allocation is set by the resource driver once a resource or set of
-    // resources has been allocated successfully. If this is not specified, the
-    // resources have not been allocated yet.
+    // Shareable indicates whether the allocated claim is meant to be shareable
+    // by multiple consumers at the same time.
     // +optional
-    Allocation *AllocationResult
+    Shareable bool
+}
 
-    // ReservedFor indicates which entities are currently allowed to use
-    // the claim. A Pod which references a ResourceClaim which is not
-    // reserved for that Pod will not be started.
+// Constraint must have one and only one field set.
+type Constraint struct {
+    // All devices must have this attribute and its value must be the same.
+    //
+    // For example, if you specified "numa.dra.example.com" (a hypothetical example!),
+    // then only devices in the same NUMA node will be chosen.
     //
-    // There can be at most 32 such reservations. This may get increased in
-    // the future, but not reduced.
     // +optional
-    ReservedFor []ResourceClaimConsumerReference
+    // +listType=atomic
+    MatchAttribute *string
+
+    // Future extension, not part of the current design:
+    // A CEL expression which compares different devices and returns
+    // true if they match.
+    //
+    // Because it would be part of a one-of, old schedulers will not
+    // accidentally ignore this additional, for them unknown match
+    // criteria.
+    //
+    // matcher string
 }
 
-// ReservedForMaxSize is the maximum number of entries in
-// claim.status.reservedFor.
-const ResourceClaimReservedForMaxSize = 32
-```
+// Request is a request for one of many resources required for a claim.
+// This is typically a request for a single resource like a device, but can
+// also ask for one of several different alternatives.
```

Comment:
```suggestion
// This is typically a request for a single resource like a device, but can
// also ask for several identical devices. It might get extended to support
// asking for one of several different alternatives.
```